bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/cache/ExpiringCacheMap.java

```diff
@@ -114,7 +108,7 @@ public void putIfAbsent(K key, ExpiringCache<V> item) {
      * @param action the action for the item to be associated with the specified key to retrieve/calculate the value
      * @return the (cached) value for the specified key
      */
-    public V putIfAbsentAndGet(K key, Supplier<V> action) {
+    public @Nullable V putIfAbsentAndGet(K key, Supplier<V> action) {
```

Comment:
It would be nice in hindsight if this `putIfAbsentAndGet` would be Nullable depending on its input :-)

Reply:
Yes, and in [line 95](https://github.com/eclipse/smarthome/pull/6482/files#diff-4a40be874170b23672abeb4314145831R95) and [line 124](https://github.com/eclipse/smarthome/pull/6482/files#diff-4a40be874170b23672abeb4314145831R124) too. Will you create a PR?

Reply:
More thoughts :-)

Given its non-synchronized nature it would be lying someone else could have invalidated the key or the complete cache. I will commit a null check for your code :-)

Reply:
Not needed anymore I see